.github/actions/draft-release/entrypoint.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
```

Comment:
Yeah it is a good question. Tbh I deliberated a bit over how best to do this and I'm not 100% confident this is the ideal approach, but I'll try to explain the tradeoffs that lead me to here:

One of the real pain points for me with mediumly complicated github actions like this is that I want to work on/test them locally as much as possible. The more I can "change file, run the script locally, see what happens" (as opposed to "change file, push to test repo, wait for the action to spin up, attempt to debug from action output") the better because it speeds up the development process. One of the things I've found with JS actions is they end up very tied to the actions API/remote repo which makes this difficult. One of the coping mechanisms I've tried employing for this (on JS actions) is trying to push as much of the logic out into utility functions which can be tested independently and keep the bits that are tied to the actions API as lean as possible but I haven't really found a workflow I love for this yet. I guess for really big action codebases it becomes viable to TDD it and mock all the external calls but it is more than I really want to do for a release script. Maybe this is partly that I just haven't found the right workflow for it. I am getting into actions but it is still early days. Anyway, meanwhile back at the point.. the nice thing about doing this as a bash script is I could develop and test almost all of this on a local checkout of the repo without having to push anything.. which made the feedback loop quicker. I suspect that will probably be a bit easier to debug in future, but I also accept we sacrifice some readability.

> I'd guess the combinations of interacting with git and not having to pull in additional parsing/process deps for the file processing?

Yes I think there are some things that are easier using shell commands. For example, I think something like `git log "$LAST_TAG"..HEAD --no-merges --oneline --pretty="format:- %s" --perl-regexp --author='^((?!dependabot).*)$'` would be quite a bit more code using the github API and javascript. That said, there are some things here that are clearly more cumbersome in bash, like templating out a JSON body to post with `curl`, so.. we probably win some and lose some. I think we're just about in the realms of sanity though.

Tbh, if we decide we want to look at identifying PRs by label or something I'd probably rewrite it in javascript. I think the pain of trying to assemble GraphQL queries in bash and extract more than a single value from the results would outweigh the pain of having to push to test :D

Reply:
> the nice thing about doing this as a bash script is I could develop and test almost all of this on a local checkout of the repo without having to push anything.. which made the feedback loop quicker. I suspect that will probably be a bit easier to debug in future, but I also accept we sacrifice some readability

Makes sense. We're not in a position where the majority of the maintainers are on Windows or anything, and from what I've seen with my work machine, WSL (particularly 2.0) would be more than sufficient in any weird edge cases where someone needed to work on it from Windows world